### DISCUSSION TITLE
Is the BMS in full control of charge input? Even external charger?

### DISCUSSION THREAD

**Discussion by optio50** [Louisvdw/dbus-serialbattery#244] [Q&A] [answered]
I had to externally charge my battery and it seems the JBD BMS does not actually control the charge input.
Is it only controlling the solar charger output?

I expected all settings and values to be respected during any kind of charging.
Current input
Voltage thresholds
Max SOC
Charge allowed

**Comment by Louisvdw** [comment, marked as answer]
You have the BMS itself that you set up with the bluetooth app. It will protect the cells and keep to those settings no matter where the charge source is except if you bypass the BMS and charge the cells directly (which you should never do)

Then we have the GX device that is part of the Victron system. The driver will make these items aware of each other and they all link to the GX and get their direction from there. So any charge from a linked item (solar MPPT, Inverter/charger, etc.) will take the directions that the battery give to the GX. Anything not linked will not know about those directions and only the protection triggers from the BMS will be used.

**Reply by optio50** [reply]
Thanks Louis
